src/blueprints/tv2/action-factories/tv2-graphics-action-factory.ts

```diff
@@ -164,4 +208,319 @@ export class Tv2GraphicsActionFactory {
+    return this.createGraphicsPieceInterface({
+      id: 'fullscreenGraphicsPiece',
+      partId,
+      name: fullscreenGraphicsData.name,
+      preRollDuration: blueprintConfiguration.studio.VizPilotGraphics.PrerollDuration,
+      pieceLifespan: this.findPieceLifespan(blueprintConfiguration, fullscreenGraphicsData.name),
+      layer: Tv2SourceLayer.PILOT_GRAPHICS,
+      content: {
+        fileName: `${PILOT_PREFIX}${fullscreenGraphicsData.vcpId}`,
+        path: `${fullscreenGraphicsData.vcpId}`,
+      },
+      timelineObjects: this.createVizFullscreenPilotGraphicsTimelineObjects(blueprintConfiguration, fullscreenGraphicsData),
+      metadata: {
+        type: Tv2PieceType.GRAPHICS,
+        outputLayer: Tv2OutputLayer.PROGRAM
+      }
+    })
+  }
+
+  private createGraphicsPartInterface(partInterfaceWithRequiredValues: Pick<PartInterface, 'id' | 'name'> & Partial<PartInterface>): PartInterface {
+    return {
+      disableNextInTransition: false,
+      expectedDuration: 0,
+      inTransition: {
+        keepPreviousPartAliveDuration: 0,
+        delayPiecesDuration: 0
+      },
+      isNext: false,
+      isOnAir: false,
+      isPlanned: false,
+      outTransition: {
+        keepAliveDuration: 0
+      },
+      pieces: [],
+      rank: 0,
+      segmentId: '',
+      isUnsynced: false,
+      ...partInterfaceWithRequiredValues
+    }
+  }
+
+  private findPieceLifespan(blueprintConfiguration: Tv2BlueprintConfiguration, templateName: string): PieceLifespan {
+    const template: GraphicsTemplate | undefined = blueprintConfiguration.showStyle.graphicsTemplates.find(
+      graphic => graphic.vizTemplate ? graphic.vizTemplate.toUpperCase() === templateName.toUpperCase() : false
+    )
+    const templateOutType: TemplateOutType = this.parseTemplateOutType(template?.outType ?? '')
+    return this.getPieceLifespanFromOutType(templateOutType)
+  }
+
+  private parseTemplateOutType(template: string): TemplateOutType {
+    switch (template) {
+      case 'S':
+        return TemplateOutType.S
+      case 'O':
+        return TemplateOutType.O
+      case 'B':
+      default:
+        return TemplateOutType.B
+    }
+  }
+
+  private getPieceLifespanFromOutType(outType: TemplateOutType): PieceLifespan {
+    switch (outType) {
+      case TemplateOutType.B:
+        return PieceLifespan.WITHIN_PART
+      case TemplateOutType.S:
+        return PieceLifespan.SPANNING_UNTIL_SEGMENT_END
+      case TemplateOutType.O:
+        return PieceLifespan.STICKY_UNTIL_RUNDOWN_CHANGE
+    }
+  }
+
+  private createVizFullscreenPilotGraphicsTimelineObjects(
+    blueprintConfiguration: Tv2BlueprintConfiguration, fullscreenGraphicsData: Tv2FullscreenGraphicsManifestData): TimelineObject[] {
+    if (!blueprintConfiguration.studio.VizPilotGraphics.CleanFeedPrerollDuration) {
+      throw new Tv2MisconfigurationException(
+        'Missing configuration of \'VizPilotGraphics.CleanFeedPrerollDuration\' in settings.'
```

Comment:
Very minor thing, but you could just use "" instead of '' inside of the string.
It's also fine to keep it as is :)

Reply:
Keeping it with single quotes. :)